File: rest/examples/python/get-parties.py
#!/usr/bin/python3

"""
Script language: Python3

Talks to:
- Vega node (REST)

Apps/Libraries:
- REST: requests (https://pypi.org/project/requests/)

Responses:
- response-examples.txt
"""

# Note: this file uses smart-tags in comments to section parts of the code to
# show them as snippets in our documentation. They are not necessary to be
# included when creating your own custom code.
#
# Example of smart-tags:
#  __something:
# some code here
# :something__

import json
import requests
import helpers

# Load REST Vega Node URL, this is set using 'source examples-config'
# located in the root folder of the api repository
node_url_rest = helpers.get_from_env("NODE_URL_REST")

# __get_parties:
# Request a list of parties trading on a Vega network
url = "{base}/parties".format(base=node_url_rest)
response = requests.get(url)
helpers.check_response(response)
response_json = response.json()
print("Parties:\n{}".format(
    json.dumps(response_json, indent=2, sort_keys=True)
))
# :get_parties__

assert len(response_json["parties"]) > 0
pubkey = response_json["parties"][0]["id"]

# __get_party_by_id:
# Request a party by their identifier (this is their public key)
url = "{base}/parties/{partyID}".format(base=node_url_rest, partyID=pubkey)
response = requests.get(url)
helpers.check_response(response)
response_json = response.json()
print("PartyById:\n{}".format(
    json.dumps(response_json, indent=2, sort_keys=True)
))
# :get_party_by_id__

File: grpc/examples/go/get-parties.go
package main

import (
	"context"
	"fmt"

	"github.com/vegaprotocol/api/grpc/clients/go/generated/code.vegaprotocol.io/vega/proto/api"
	"github.com/vegaprotocol/api/grpc/examples/go/helpers"

	"google.golang.org/grpc"
)

func main() {
	nodeURLGrpc := helpers.GetFromEnv("NODE_URL_GRPC")
	conn, err := grpc.Dial(nodeURLGrpc, grpc.WithInsecure())
	if err != nil {
		panic(err)
	}
	defer conn.Close()

	dataClient := api.NewTradingDataServiceClient(conn)

	// __get_parties:
	// Request a list of parties trading on a Vega network
	request := api.PartiesRequest{}
	parties, err := dataClient.Parties(context.Background(), &request)
	if err != nil {
		panic(err)
	}

	fmt.Printf("Parties: %v\n", parties)
	// :get_parties__

	partyID := parties.Parties[0].Id
	// __get_party_by_id:
	// Request a party by their identifier (this is their public key)
	partyByIDReq := api.PartyByIDRequest{PartyId: partyID}
	partyByIDResp, err := dataClient.PartyByID(context.Background(), &partyByIDReq)
	if err != nil {
		panic(err)
	}

	fmt.Printf("PartyById: %v\n", partyByIDResp.Party)
	// :get_party_by_id__
}
